proposals/lip-0061.md

```diff
@@ -517,14 +520,18 @@ An object following the schema `aggregateCommitSchema`, which can be added the n
 
 ```python
 def selectAggregateCommit() -> AggregateCommit:
-    # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated by the previous certificate.
-    heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified + 1)
-    if previous function call returns valid height:
-        nextHeight = min(heightNextBFTParameters-1, getBFTHeights().maxHeightPrecommitted)
-    elif previous function call returns Not Found error:
+    try:
+        # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated by the previous certificate.
+        heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified + 1)
+        nextHeight = min(heightNextBFTParameters -1, getBFTHeights().maxHeightPrecommitted)
+    except:  
+        # getNextHeightBFTParameters function call returns error, i.e., there are no new bft params since last certificate.
+        # attempt to create certificates for finalized blocks
         nextHeight = getBFTHeights().maxHeightPrecommitted
 
-    while nextHeight > getBFTHeights().maxHeightCertified:
+    nextHeight = max(nextHeight, MIN_CERTIFICATE_HEIGHT)    
+
+    while nextHeight > max(getBFTHeights().maxHeightCertified, MIN_CERTIFICATE_HEIGHT - 1):
```

Comment:
`MIN_CERTIFICATE_HEIGHT` by default should be `1` in this case?

Reply:
Can't it be 0?

Reply:
I think it makes more sense if the default is `1`, but probably doesnt change much

Reply:
It does not make sense to generate certificates for the genesis block so I would also set it to 1 by default.

Reply:
Ok switched the definition to genesis height + 1.

Reply:
Can be resolved from my side.